Arm/rover_arm_system.hpp

```diff
@@ -3,58 +3,199 @@
 #include "Hand/wrist_joint.hpp"
 #include "../Common/rover_system.hpp"
+#include <cmath>
 
 namespace sjsu::arm
 {
 class RoverArmSystem : public sjsu::common::RoverSystem
 {
- private:
-  // isOperational defines if the arm should be allowed to move or if it should
-  // be in its resting 'off' position.
-  bool isOperational;
-
-  sjsu::arm::Joint & Rotunda;
-  sjsu::arm::Joint & Shoulder;
-  sjsu::arm::Joint & Elbow;
-  sjsu::arm::WristJoint & Wrist;
-
-  // The target angle for each of the joints
-  units::angle::degree_t shoulder_pos;
-  units::angle::degree_t elbow_pos;
-  units::angle::degree_t rotunda_pos;
-  units::angle::degree_t wrist_roll_pos;
-  units::angle::degree_t wrist_pitch_pos;
-
  public:
+  struct MissionControlData
+  {
+    enum class Modes : char
+    {
+      kDefault = 'D',
+    };
+    int is_operational;
+    int heartbeat_count = 0;
+    Modes modes         = Modes::kDefault;
+    int rotunda_speed;
+    int rotunda_angle;
+    int shoulder_speed;
+    int shoulder_angle;
+    int elbow_speed;
+    int elbow_angle;
+    int wrist_speed;
+    int wrist_roll;
+    int wrist_pitch;
+    struct Fingers
+    {
+      int pinky;
+      int ring;
+      int middle;
+      int pointer;
+      int thumb;
+    };
+    Fingers fingers;
+  };
+
   RoverArmSystem(sjsu::arm::Joint & rotunda,
                  sjsu::arm::Joint & shoulder,
                  sjsu::arm::Joint & elbow,
                  sjsu::arm::WristJoint wrist)
-      : Rotunda(rotunda), Shoulder(shoulder), Elbow(elbow), Wrist(wrist)
+      : rotunda_(rotunda), shoulder_(shoulder), elbow_(elbow), wrist_(wrist)
   {
   }
   void Initialize()
   {
-    Rotunda.Initialize();
-    Shoulder.Initialize();
-    Elbow.Initialize();
-    Wrist.Initialize();
+    rotunda_.Initialize();
+    shoulder_.Initialize();
+    elbow_.Initialize();
+    wrist_.Initialize();
   }
 
   void PrintRoverData(){};
 
-  std::string GETParameters(){};
+  std::string GETParameters()
+  {
+    return "Fill";
+  };
 
-  std::string ParseJSONResponse(){};
+  std::string ParseJSONResponse()
+  {
+    return "Fill";
+  };
 
-  void HandleRoverMovement(){};
+  void HandleArmMovement()
+  {
+    // rotunda
+    units::angular_velocity::revolutions_per_minute_t rotunda_speed(
+        static_cast<float>(mc_data.rotunda_speed));
+    units::angle::degree_t rotunda_angle(
+        static_cast<float>(mc_data.rotunda_angle));
+    // shoulder
+    units::angular_velocity::revolutions_per_minute_t shoulder_speed(
+        static_cast<float>(mc_data.shoulder_speed));
+    units::angle::degree_t shoulder_angle(
+        static_cast<float>(mc_data.shoulder_angle));
+    // elbow
+    units::angular_velocity::revolutions_per_minute_t elbow_speed(
+        static_cast<float>(mc_data.elbow_speed));
+    units::angle::degree_t elbow_angle(static_cast<float>(mc_data.elbow_angle));
+    // wrist
+    units::angular_velocity::revolutions_per_minute_t wrist_speed(
+        static_cast<float>(mc_data.wrist_speed));
```

Comment:
all this casting could be avoided - see above. This logic should be done in the implementation class imo. It clutters up this code base a bit too much